network/src/protocols/network/mod.rs

```diff
@@ -69,6 +71,50 @@ impl<TMessage: PartialEq> PartialEq for Event<TMessage> {
     }
 }
 
+/// Configuration needed for DiemNet applications to register with the network
+/// builder. Supports client-only, service-only, and p2p (both) applications.
+// TODO(philiphayes): separate configs for client & server?
+#[derive(Clone, Default)]
+pub struct AppConfig {
+    /// The set of protocols needed for this application.
+    pub protocols: ProtocolIdSet,
+    /// The config for the inbound message queue from network to the application.
+    /// Used for specifying the queue style (e.g. FIFO vs LIFO) and sub-queue max
+    /// capacity.
+    // TODO(philiphayes): only relevant for services
+    // TODO(philiphayes): in the future, use a Service trait here instead?
+    pub inbound_queue: Option<diem_channel::Config>,
```

Comment:
Let's make a followup github issue that lists out the things missing on the path to client / server model.  This is a great place to start with it!